2021/Robot2021/src/main/cpp/subsystems/Drivetrain.cpp

```diff
@@ -626,6 +602,46 @@ degree_t Drivetrain::GetHeadingAngle()
     }
 }
 
+meter_t Drivetrain::GetDistanceMetersLeft()
+{
+    if (frc::RobotBase::IsReal())
+    {
+        return kEncoderMetersPerCount * m_motorL1.GetSelectedSensorPosition(kPidIndex);
+    }
+    else
+    {
+        return m_leftEncoder.GetDistance() * 1_m;
+    }
+}
+
+meter_t Drivetrain::GetDistanceMetersRight()
+{
+    if (frc::RobotBase::IsReal())
+    {
+        return kEncoderMetersPerCount * m_motorR3.GetSelectedSensorPosition(kPidIndex);
+    }
+    else
+    {
+        return m_rightEncoder.GetDistance() * 1_m;
+    }
+}
+
+frc::DifferentialDriveWheelSpeeds Drivetrain::GetRateMPS() 
+{
+    meters_per_second_t leftVelocity;
+    meters_per_second_t rightVelocity;
+
+    if (frc::RobotBase::IsReal()) {
+        leftVelocity = kEncoderMetersPerCount * m_motorL1.GetSelectedSensorVelocity() * 10 / 1_s;
+        rightVelocity = kEncoderMetersPerCount * m_motorR3.GetSelectedSensorVelocity() * 10 / 1_s;
+    }
+    else {
+        leftVelocity = m_leftEncoder.GetRate() * 1_mps;
+        rightVelocity = m_rightEncoder.GetRate() * 1_mps;

```

Comment:
Everything looks organized really well and succinct, I can't find anything you could change in your code :)

Reply:
Agree with Christine. The formatting looks really good!